docs/UserGuide.md

```diff
@@ -40,79 +40,233 @@ Format: `help`
+*`add <task description> on/at/by <date> <time>`
 
-> Persons can have any number of tags (including 0)
+> A timed task has a specified completion date and/or time. 
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe p/1234567 e/betsycrowe@gmail.com a/Newgate Prison t/criminal t/friend`
+* `add finish homework by 4pm today`
+* `add buy shoes on Sep 16 6:30pm`
+* `add call grandmother at 15:00 on Oct 17`
 
-#### Listing all persons : `list`
-Shows a list of all persons in the address book.<br>
-Format: `list`
+Adds an untimed task<br>
+Format: 
+*`add <task description>` 
 
-#### Finding all persons containing any keyword in their name: `find`
-Finds persons whose names contain any of the given keywords.<br>
-Format: `find KEYWORD [MORE_KEYWORDS]`
+> An untimed task doesn't have a specified completion date and time.
+
+Examples: 
+* `add buy grocery recently`
+* `add be careful across the road`
+
+Adds a recurring task<br>
+Format: 
+*`add <task description> each <calendar day> at/by <time> ^<no. of weeks the task recurs>` 
+
+> A recurring task takes place each calendar day(s) for the specified number of weeks. 
+
+Examples: 
+* `add watch movie each Friday ^5`
+* `add play basketball each Tue and each Sat at 5pm ^7`
+
+Adds hashtag<br>
+Format: 
+*`#<hashtag>` 
+
+> To add hashtag(s) to task entries. 
+
+Examples: 
+* `add buy grocery at 7pm #chore`
+
+
+#### Editing a task: `edit`
+Edits a task already in the list<br>
+Format: 
+*`edit <index> <new description> at/on/by <new date/time> #<new hashtags>`
+
+> New input will replace old task entry completely.   
+
+Examples: 
+* `edit 1 buy grocery at 5pm`
+
+
+#### Marking of a task as done: `done`
+Marks a completed task.<br>
+Format: `done INDEX`
+
+> Marks the task entry as done at the specified `INDEX`. 
+  The index refers to the index number shown in the most recent listing.<br>
+  The index **must be a positive integer** 1, 2, 3, ...
+
+Examples: 
+* `list`<br>
+  `done 2`<br>
+  Marks the 2nd task entry in all the undone tasks listed as done.
+* `find #grocery`<br> 
+  `done 1`<br>
+  Marks the 1st task entry in the results of the `find` command as done.
+  
+#### Marking of a task as undone: `undone`
+Unmark a completed task.<br>
+Format: `undone INDEX`
+
+> Marks a task entry as undone at the specified `INDEX`. 
+  The index refers to the index number shown in the completed task listing.<br>
+  The index **must be a positive integer** 1, 2, 3, ...
+
+Examples: 
+* `done 2`<br>
+  Marks the 2nd task entry in the completed task listing as undone.
+   
+  
+#### Listing of tasks : `list`
+Lists all tasks<br>
+Format: 
+*`list all`
+*`list`
+
+> Lists all undone tasks. 
+
+
+Lists untimed tasks<br>
+Format: 
+*`list untimed` 
+
+> Lists all untimed tasks. 
+
+
+Lists timed tasks<br>
+Format: 
+*`list timed` 
+
+> Lists all timed tasks. 
+
+
+Lists tasks with hashtags<br>
+Format: 
+*`list #<keyword>` 
```

Comment:
"keyword" should be "hashtag"? And what is the difference between the `find #hashtag` ([line 110](https://github.com/nus-cs2103-AY1617S1/addressbook-level4/pull/86/files#diff-572f9bedcb201b96c74241fb42e29fcfR110)) and `list #keyword` here?